src/models/section.ts

```diff
@@ -11,15 +11,16 @@ export enum SectionType {
 }
 
 export const sectionInfo = {
-  [SectionType.introduction]: { title: "Introduction", abbrev: "In" },
-  [SectionType.initialChallenge]: { title: "Initial Challenge", abbrev: "IC" },
-  [SectionType.whatIf]: { title: "What if...?", abbrev: "W?" },
-  [SectionType.nowWhatDoYouKnow]: { title: "Now What Do You Know?", abbrev: "N?" },
-  [SectionType.didYouKnow]: { title: "Did You Know?", abbrev: "D?" }
+  [SectionType.introduction]: { id: SectionType.introduction, title: "Introduction", abbrev: "In" },
+  [SectionType.initialChallenge]: { id: SectionType.initialChallenge, title: "Initial Challenge", abbrev: "IC" },
+  [SectionType.whatIf]: { id: SectionType.whatIf, title: "What if...?", abbrev: "W?" },
+  [SectionType.nowWhatDoYouKnow]: { id: SectionType.nowWhatDoYouKnow, title: "Now What Do You Know?", abbrev: "N?" },
+  [SectionType.didYouKnow]: { id: SectionType.didYouKnow, title: "Did You Know?", abbrev: "D?" }
 };
 
 export const SectionModel = types
   .model("Section", {
+    id: types.identifier,
```

Comment:
Identifiers are required to be unique within the entire tree, so this isn't going to work as an MST identifier. In combination with the reference comment, I think it would make sense to just have the reference be a string which could indicate the type which we dereference in code.
> Each identifier / type combination should be unique within the entire tree

Reply:
This does load the json now and works fine as is because we never have more than 1 problem in the tree and thus there are no duplicate section ids.  I think we should merge and then discuss references after the demo.